metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java

```diff
@@ -0,0 +1,254 @@
+    final List<RelationshipFieldSpec> optionalRelationshipFieldSpec =
+        aspectSpec.findRelationshipFor(relationType, urn.getEntityType());
+
+    if (optionalRelationshipFieldSpec.isEmpty()) {
+      log.error("Unable to find relationship spec information in {} that connects to {} via {}",
+          aspectName, urn.getEntityType(), relationType);
+      return;
+    }
+
+    for (RelationshipFieldSpec relationshipFieldSpec : optionalRelationshipFieldSpec) {
+      final PathSpec path = relationshipFieldSpec.getPath();
+      final Aspect updatedAspect = DeleteReferencesService.getAspectWithReferenceRemoved(urn.toString(), aspect,
+          aspectSpec.getPegasusSchema(), path);
+
+      final MetadataChangeProposal gmce = new MetadataChangeProposal();
+      gmce.setEntityUrn(relatedUrn);
+      gmce.setChangeType(ChangeType.UPSERT);
+      gmce.setEntityType(relatedEntityName);
+      gmce.setAspectName(aspectName);
+      gmce.setAspect(GenericRecordUtils.serializeAspect(updatedAspect));
+
+      final AuditStamp auditStamp = new AuditStamp().setActor(UrnUtils.getUrn(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
+      final EntityService.IngestProposalResult ingestProposalResult = _entityService.ingestProposal(gmce, auditStamp);
+
+      if (!ingestProposalResult.isDidUpdate()) {
+        log.warn("Aspect update did not update metadata graph. Before {}, after: {}", aspect, updatedAspect);
+      }
+    }
+  }
+
+  /**
+   * Utility method that attempts to find Aspect information as well as the associated path spec for a given urn that
+   * has a relationship of type `relationType` to another urn
+   * @param urn               The urn of the entity for which we want to find the aspect that relates the urn to
+   *                          `relatedUrn`.
+   * @param relatedUrn        The urn of the related entity in which we want to find the aspect that has a relationship
+   *                          to `urn`.
+   * @param relationType      The relationship type that details how the `urn` entity is related to `relatedUrn` entity.
+   * @param relatedEntitySpec The entity spec of the related entity.
+   * @return An {@link java.util.Optional} object containing the aspect content & respective spec that contains the
+   * relationship between `urn` & `relatedUrn`.
+   */
+  public java.util.Optional<Pair<EnvelopedAspect, AspectSpec>> getAspectReferringTo(final Urn urn, final Urn relatedUrn,
+      final String relationType, final @Nonnull EntitySpec relatedEntitySpec) {
+
+    // Find which aspects are the candidates for the relationship we are looking for
+    final Map<String, AspectSpec> aspectSpecs = relatedEntitySpec.getAspectSpecMap()
+        .entrySet()
+        .stream()
+        .filter(aspectSpec -> !aspectSpec.getValue().findRelationshipFor(relationType, urn.getEntityType()).isEmpty())
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
on the aspect spec, maybe we can just ask for all relationship types contained by the aspect - "getRelationshipTypes()" .. and then say if aspectSpec.getRelationshipTypes().contains(relationship) // do something